### PR TITLE
Update Regridding notebook to to work with newest xarray/xesmf

### LINKED ISSUE #258: regridding.ipynb broken with upcoming Jupyter env on PAVICS
Upcoming Jupyter env is the `beta` env on PAVICS.

Opened branch in this repo for upcoming Jupyter env, commit here https://github.com/Ouranosinc/pavics-sdi/tree/fix-notebooks-for-upcoming-jupyter-env-release

http://jenkins.ouranos.ca/job/PAVICS-e2e-workflow-tests/job/new-docker-build/80/console

 ValueError: dimension 'time' already exists as a scalar variable

22:12:46  _______ pavics-sdi-master/docs/source/notebooks/regridding.ipynb::Cell 7 _______
22:12:46  Notebook cell execution failed
22:12:46  Cell 7: Cell execution caused an exception
22:12:46  
22:12:46  Input:
22:12:46  # xesmf/frontend.py:476: FutureWarning: ``output_sizes`` should be given in the ``dask_gufunc_kwargs`` parameter. It will be removed as direct parameter in a future version.
22:12:46  warnings.filterwarnings("ignore", category=FutureWarning)
22:12:46  
22:12:46  # Apply the regridding weights to the input sea ice concentration data
22:12:46  sic_bil = reg_bil(ds_in.siconc)
22:12:46  
22:12:46  # Plot the results
22:12:46  sic_bil.isel(time=0).plot(cmap=cmap)
22:12:46  plt.title('Regridded sic data (Jan 2020)');
22:12:46  
22:12:46  Traceback:
22:12:46  
22:12:46  ---------------------------------------------------------------------------
22:12:46  ValueError                                Traceback (most recent call last)
22:12:46  Input In [8], in <cell line: 5>()
22:12:46        2 warnings.filterwarnings("ignore", category=FutureWarning)
22:12:46        4 # Apply the regridding weights to the input sea ice concentration data
22:12:46  ----> 5 sic_bil = reg_bil(ds_in.siconc)
22:12:46        7 # Plot the results
22:12:46        8 sic_bil.isel(time=0).plot(cmap=cmap)
22:12:46  
22:12:46  File /opt/conda/envs/birdy/lib/python3.8/site-packages/xesmf/frontend.py:458, in BaseRegridder.__call__(self, indata, keep_attrs, skipna, na_thres)
22:12:46      456     return self.regrid_array(indata, self.weights.data, skipna=skipna, na_thres=na_thres)
22:12:46      457 elif isinstance(indata, xr.DataArray):
22:12:46  --> 458     return self.regrid_dataarray(
22:12:46      459         indata, keep_attrs=keep_attrs, skipna=skipna, na_thres=na_thres
22:12:46      460     )
22:12:46      461 elif isinstance(indata, xr.Dataset):
22:12:46      462     return self.regrid_dataset(
22:12:46      463         indata, keep_attrs=keep_attrs, skipna=skipna, na_thres=na_thres
22:12:46      464     )
22:12:46  
22:12:46  File /opt/conda/envs/birdy/lib/python3.8/site-packages/xesmf/frontend.py:549, in BaseRegridder.regrid_dataarray(self, dr_in, keep_attrs, skipna, na_thres)
22:12:46      537 kwargs = dict(skipna=skipna, na_thres=na_thres)
22:12:46      538 dr_out = xr.apply_ufunc(
22:12:46      539     self.regrid_array,
22:12:46      540     dr_in,
22:12:46     (...)
22:12:46      546     keep_attrs=keep_attrs,
22:12:46      547 )
22:12:46  --> 549 return self._format_xroutput(dr_out, temp_horiz_dims)
22:12:46  
22:12:46  File /opt/conda/envs/birdy/lib/python3.8/site-packages/xesmf/frontend.py:830, in Regridder._format_xroutput(self, out, new_dims)
22:12:46      826 if new_dims is not None:
22:12:46      827     # rename dimension name to match output grid
22:12:46      828     out = out.rename({nd: od for nd, od in zip(new_dims, self.out_horiz_dims)})
22:12:46  --> 830 out = out.assign_coords(**self.out_coords)
22:12:46      831 out.attrs['regrid_method'] = self.method
22:12:46      833 if self.sequence_out:
22:12:46  
22:12:46  File /opt/conda/envs/birdy/lib/python3.8/site-packages/xarray/core/common.py:592, in DataWithCoords.assign_coords(self, coords, **coords_kwargs)
22:12:46      590 data = self.copy(deep=False)
22:12:46      591 results: dict[Hashable, Any] = self._calc_assign_results(coords_combined)
22:12:46  --> 592 data.coords.update(results)
22:12:46      593 return data
22:12:46  
22:12:46  File /opt/conda/envs/birdy/lib/python3.8/site-packages/xarray/core/coordinates.py:165, in Coordinates.update(self, other)
22:12:46      161 self._maybe_drop_multiindex_coords(set(other_vars))
22:12:46      162 coords, indexes = merge_coords(
22:12:46      163     [self.variables, other_vars], priority_arg=1, indexes=self.xindexes
22:12:46      164 )
22:12:46  --> 165 self._update_coords(coords, indexes)
22:12:46  
22:12:46  File /opt/conda/envs/birdy/lib/python3.8/site-packages/xarray/core/coordinates.py:376, in DataArrayCoordinates._update_coords(self, coords, indexes)
22:12:46      374 coords_plus_data = coords.copy()
22:12:46      375 coords_plus_data[_THIS_ARRAY] = self._data.variable
22:12:46  --> 376 dims = calculate_dimensions(coords_plus_data)
22:12:46      377 if not set(dims) <= set(self.dims):
22:12:46      378     raise ValueError(
22:12:46      379         "cannot add coordinates with new dimensions to a DataArray"
22:12:46      380     )
22:12:46  
22:12:46  File /opt/conda/envs/birdy/lib/python3.8/site-packages/xarray/core/variable.py:3066, in calculate_dimensions(variables)
22:12:46     3064 for dim, size in zip(var.dims, var.shape):
22:12:46     3065     if dim in scalar_vars:
22:12:46  -> 3066         raise ValueError(
22:12:46     3067             f"dimension {dim!r} already exists as a scalar variable"
22:12:46     3068         )
22:12:46     3069     if dim not in dims:
22:12:46     3070         dims[dim] = size
22:12:46  
22:12:46  ValueError: dimension 'time' already exists as a scalar variable

### PR DESCRIPTION
**Pull request by Zeitsperre**
Closes #258 

For new Jupyter env in https://github.com/Ouranosinc/PAVICS-e2e-workflow-tests/pull/106.

### CODE REVIEW

**Review by tlvu — APPROVED** [review]
Fix looks good.

Should wait and merge this PR only when the new Jupyter env go-live.  Also need to sync this merge with https://github.com/bird-house/finch/issues/253

**Comment by huard** [comment]
What's the status on this?

**Comment by Zeitsperre** [comment]
Not yet ready. @aulemahal has changes that he's going to push to get the regridding notebook working again with the newest xarray.

**Comment by aulemahal** [comment]
My flu has shifted the task towards @tlogan2000 I think.

**Comment by tlogan2000** [comment]
I'll fix after my meeting at 10 am.  I suggest a jenkins run before merging

**Comment by tlogan2000** [comment]
Changing the order of the imports (geopandas / owslib first) seems to fix this bug of the kernel dying... Not sure this is a great fix but is all I can really do short term.   I will fire off a jenkins build to test before merging

**Comment by Zeitsperre** [comment]
@tlogan2000 

This will help: 
https://pycqa.github.io/isort/docs/configuration/action_comments.html#isort-skip

**Comment by tlogan2000** [comment]
Notebooks seem to be passing (failures all seem to be simply differences in the outputs) http://jenkins.ouranos.ca/job/PAVICS-e2e-workflow-tests/job/new-docker-build/100.

**Comment by tlvu** [comment]
> Notebooks seem to be passing (failures all seem to be simply differences in the outputs) http://jenkins.ouranos.ca/job/PAVICS-e2e-workflow-tests/job/new-docker-build/100.

Great so you or someone will simply refresh the failing noteobooks?  I'd like someone else to do it other than me so the team is resilient.

Remember to use the matching Jupyter env to refresh the notebooks so the output matches.  Also use the option "Restart kernel and run all cells" to refresh the notebooks to ensure a clean run.

**Comment by tlvu** [comment]
This notebook PR and probably a matching one on Finch side to refresh the notebooks will need to be merged in sync with the go-live of the new matching Jupyter env.

Basically let me do the merge of all notebooks PR so I sync it with the go-live of the new Jupyter env.

**Comment by tlogan2000** [comment]
> > Notebooks seem to be passing (failures all seem to be simply differences in the outputs) http://jenkins.ouranos.ca/job/PAVICS-e2e-workflow-tests/job/new-docker-build/100.
> 
> Great so you or someone will simply refresh the failing noteobooks? I'd like someone else to do it other than me so the team is resilient.
> 
> Remember to use the matching Jupyter env to refresh the notebooks so the output matches. Also use the option "Restart kernel and run all cells" to refresh the notebooks to ensure a clean run.

I'll try to get to the refresh today

**Comment by tlogan2000** [comment]
@tlvu  the following branches should refresh notebooks from finch and pavics landing:
https://github.com/bird-house/finch/tree/refresh_notebooks
https://github.com/Ouranosinc/PAVICS-landing/tree/refresh_notebooks

Rerunning jenkins here : http://jenkins.ouranos.ca/job/PAVICS-e2e-workflow-tests/job/new-docker-build/101/

File: docs/source/notebooks/regridding.ipynb
Comment (tlvu) [inline docs/source/notebooks/regridding.ipynb:1123]: @tlogan2000 this was the stderr that tripped Jenkins.  It's weird Jenkins do not have this but when we "restart kernel and run all cells" we have this.

**Comment by tlvu** [comment]
This PR merged into PR https://github.com/Ouranosinc/pavics-sdi/pull/267

**Pull request merged**
